sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py

```diff
@@ -827,6 +829,89 @@ async def peek_messages(
+        :keyword datetime.datetime or None before_enqueued_time_utc: The UTC datetime value before which all messages
+         should be deleted. The default value is None, meaning all messages in the queue will be considered.
+        :keyword Optional[float] timeout: The total operation timeout in seconds including all the retries.
+         The value must be greater than 0 if specified. The default value is None, meaning no timeout.
+        :return: The number of deleted messages.
+        :rtype: int
+
+        """
+        self._check_live()
+        if timeout is not None and timeout <= 0:
+            raise ValueError("The timeout must be greater than 0.")
+        await self._open()
+
+        message_count = max_message_count if max_message_count else 4000
+
+        message = {
+            MGMT_REQUEST_ENQUEUED_TIME_UTC: before_enqueued_time_utc if before_enqueued_time_utc
+                else datetime.datetime.now(datetime.timezone.utc),
+            MGMT_REQUEST_MAX_MESSAGE_COUNT: message_count,
+        }
+
+        self._populate_message_properties(message)
+        handler = functools.partial(mgmt_handlers.batch_delete_op, receiver=self, amqp_transport=self._amqp_transport)
+        deleted = await self._mgmt_request_response_with_retry(
+            REQUEST_RESPONSE_DELETE_BATCH_OPERATION, message, handler, timeout=timeout
+        )
+
+        return deleted
+
+    async def purge_messages(
+        self,
+        *,
+        before_enqueued_time_utc: Optional[datetime.datetime] = None,
+        timeout: Optional[float] = None,
+    ) -> int:
+        """
+        This operation purges as many messages as possible in the queue that are older than the specified enqueued time.
+
+        :keyword datetime.datetime or None before_enqueued_time_utc: The UTC datetime value before which all messages
+         should be deleted. The default value is None, meaning all messages from the current time and before
+         in the queue will be considered.
+        :keyword Optional[float] timeout: The total operation timeout in seconds including all the retries.
+         The value must be greater than 0 if specified. The default value is None, meaning no timeout.
+        :return: The number of deleted messages.
+        :rtype: int
+
+        """
+        self._check_live()
+        if timeout is not None and timeout <= 0:
+            raise ValueError("The timeout must be greater than 0.")
+        await self._open()
+
+        message = {
+            MGMT_REQUEST_ENQUEUED_TIME_UTC: before_enqueued_time_utc if before_enqueued_time_utc
+                else datetime.datetime.now(datetime.timezone.utc),
+            MGMT_REQUEST_MAX_MESSAGE_COUNT: 4000,
+        }
+
+        self._populate_message_properties(message)
+        handler = functools.partial(mgmt_handlers.batch_delete_op, receiver=self, amqp_transport=self._amqp_transport)
+
+        batch_count = 0
+        deleted = None
+        while deleted != 0:
```

Comment:
doing this instead of == max_message_count because service will not guarantee that we will get back the max